docs(datasets): add the extra install instructions (#669)
kedro-org/kedro-plugins: kedro-datasets/README.md

pip install kedro-datasets
```

> ### Install dependencies at a group-level

Author: ```suggestion
### Install dependencies at a group level
```
Not sure why this is hyphenated, but I blamed the original source, and I it was added by @stichbury, so who am I to question the grammar. :joy:

Reviewer: If @stichbury says it's hyphenated, it's hyphenated 🙇🏼

Reviewer 2: Thanks! Although I'm not sure I'm right here, but there would have been a reason and it's consistent with the next heading, so let's keep it!

> Data types are broken into groups e.g. `pandas`, `spark` and `pickle`. Each group has a collection of data types e.g.`pandas.CSVDataset`, `pandas.ParquetDataset` and more. You can install dependencies for an entire group of dependencies as follows:

Author: I do find the term "data types" a bit weird, when we call them "datasets" everywhere else.

Reviewer: Yes let's call them datasets.

```suggestion
Datasets are organised into groups e.g. `pandas`, `spark` and `pickle`. Each group has a collection of datasets, e.g.`pandas.CSVDataset`, `pandas.ParquetDataset` and more. You can install dependencies for an entire group of dependencies as follows:
```

?

```bash
pip install "kedro-datasets[<group>]"
```

> This installs Kedro and dependencies related to the data type group. An example of this could be a workflow that depends on the data types in `pandas`. Run `pip install "kedro-datasets[pandas]"` to install Kedro and the dependencies for the data types in the [`pandas` group](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets/kedro_datasets/pandas).

Reviewer: ```suggestion
This installs `kedro-datasets` and dependencies related to the dataset group. An example of this could be a workflow that depends on the data types in `pandas`. Run `pip install "kedro-datasets[pandas]"` to install `kedro-datasets` and the dependencies for the datasets in the [`pandas` group](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets/kedro_datasets/pandas).
```

Author: (Changed `kedro-datasets` to Kedro-Datasets; more in line with the Kedro lexicon guide, and I'll do more standardization after this.)

> ### Install dependencies at a type-level

Author: ```suggestion
### Install dependencies at a type level
```
Same comment as https://github.com/kedro-org/kedro-plugins/pull/669/files#r1593140715.

Reviewer: Same response

> To limit installation to dependencies specific to a data type:

Reviewer: ```suggestion
To limit installation to dependencies specific to a dataset:
```

```bash

> pip install "kedro-datasets[<group>-<dataset>]"

Reviewer: ```suggestion
pip install "kedro-datasets[<group>-<lowercase-dataset>]"
```

?

Author: Extra names should be normalized already; think this is unnecessary unless start getting issues.

Ref: https://peps.python.org/pep-0685/

```

For example, your workflow might require the `pandas.ExcelDataset`, so to install its dependencies, run `pip install "kedro-datasets[pandas-exceldataset]"`.

```{note}

> From `kedro-datasets` version 3.0.0 onwards, the names of the optional dataset-level dependencies have been normalised to follow [PEP 685](https://peps.python.org/pep-0685/). The '.' character has been replaced with a '-' character and the names are in lowercase. For example, if you had `kedro-datasets[pandas.ExcelDataset]` in your requirements file, it would have to be changed to `kedro-datasets[pandas-exceldataset]`.

Reviewer: 💯

Author: Oh, there's already a PEP 685 ref here. :joy:

This is wrong, though? Tools writing core metadata (i.e. people publishing packages) should write in the normalized form. Users can still do `pandas.ExcelDataSet`, no problem (and I like this better).

Author: I kinda wanna revert this stuff back to the good old days. Let me `blame` and figure out who to confer with.
